File: littleware/src/main/java/littleware/test/LittleTest.java
package littleware.test;

import java.util.logging.Level;
import java.util.logging.Logger;
import static org.junit.Assert.*;

/**
 * Some utilities for tests
 */
public final class LittleTest {
    /** Every test wants a logger */
    public static final Logger log = Logger.getLogger( LittleTest.class.getName() );
    
    /**
     * Typical exception handler
     */
     public static void handle( Throwable ex ) {
         log.log( Level.WARNING, "Failed test", ex );
         fail( "Caught: " + ex );
     } 
   
}

File: littleScala/src/main/scala/littleware/scala/test/LittleTest.scala
package littleware.scala.test


/**
 * Little specialization of littleware.test.LittleTest with
 * some scala friendliness
 */
class LittleTest {
  val basicHandler:PartialFunction[Throwable,Unit] = {
    case ex:Exception => littleware.test.LittleTest.handle(ex)
    case ex:java.lang.AssertionError => littleware.test.LittleTest.handle(ex)
    case ex => throw ex
  }
}
